### PR TITLE
UI: product area / product overview page

### PR DESCRIPTION
**Pull request by tobbber**
See Issue #48  

**Basic Layout**
- [x] Page with title of product area
- [x] Table/List of Products in product area.
- [x] Button "Edit"
- [x] Each Table/List Item: (Product Name, Button economic rating, Button komplexity rating, Text field Notes/Comments, edit buttons)
maybe add additional field `variant of`, this will later be replaced by the layout

**Mock**
- [ ] add Product area mock to display. Model this Mock after the data returned by the API

**Functional** (reuse as much logic as possible from manage project page)
- [x] Edit button: Enter edit mode
- [x] If not in edit mode: hide buttons for editing products.
- [x] Edit mode: Button for adding Adding new products to the list (eg. "a plus button on the table header / footer")
- [ ] Edit mode: Button for adding Adding new variants to a product (eg. "a plus button next to each product")
- [x] Edit mode: Remove button for products and product variants
- [x] Confirm/Cancel buttons to apply/discard changes made in edit mode
- [x] Enter rating buttons lead to empty site showing bsp rating or product ID

**Dynamic**

- [ ] get data for project area from backend
- [ ] update products if changes confirmed

### CODE REVIEW

File: frontend/quickcheck/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "react-dom": "^17.0.2",
     "react-router-dom": "^6.0.2",
     "react-scripts": "4.0.3",
+    "uuid4": "^2.0.2",
```
Comment (tobbber) [inline frontend/quickcheck/package.json:22]: I dont think we need this dependency

File: frontend/quickcheck/src/components/ProductElement.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+                    als button text vielleicht den bewertungsfortschritt (zb. props.product.rating.economic)
+                - button für komplexitätsbewertung
+                - textfeld für anmerkungen
+                - button für nachweise mit text "upload" (vielleicht iconbutton mit schönem icon :) 
+            */}
+    </HStack>
+  );
+}
+
+export default ProductElement;
```
Comment (tobbber) [inline frontend/quickcheck/src/components/ProductElement.jsx:19]: Is this file needed? if not, remove it

File: frontend/quickcheck/src/components/ProjectElement.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+                    als button text vielleicht den bewertungsfortschritt (zb. props.product.rating.economic)
+                - button für komplexitätsbewertung
+                - textfeld für anmerkungen
+                - button für nachweise mit text "upload" (vielleicht iconbutton mit schönem icon :) 
+            */}
+    </HStack>
+  );
+}
+
+export default ProjectElement;
```
Comment (tobbber) [inline frontend/quickcheck/src/components/ProjectElement.jsx:19]: Is this fille used?

Comment (aron4) [reply frontend/quickcheck/src/components/ProjectElement.jsx:19]: No, will be deleted.

File: frontend/quickcheck/src/pages/ProjectOverview.jsx [hunk 4]
```diff
@@ -8,17 +8,12 @@ import { Link } from 'react-router-dom';
```
Comment (tobbber) [inline frontend/quickcheck/src/pages/ProjectOverview.jsx:8]: These Changes break the Add project button, i think this is an old version of this file.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "@chakra-ui/icons": "^1.1.1"
```
Comment (tobbber) [inline package.json:3]: I dont think this file is needed

Comment (aron4) [reply package.json:3]: Don't we use the AddIcon and DeleteIcon from @chakra-ui/icons?

Comment (tobbber) [reply package.json:3]: Yes but its already installed just version 1.2.1 (see yarn.lock file)

Comment (aron4) [reply package.json:3]: Alright, will remove it

File: frontend/quickcheck/src/pages/ProjectOverview.jsx (hunk 4 above)
Comment (aron4) [reply frontend/quickcheck/src/pages/ProjectOverview.jsx:8]: How should this be resolved?

Comment (tobbber) [reply frontend/quickcheck/src/pages/ProjectOverview.jsx:8]: `git checkout origin/development ProjectOverview.jsx` should do the trick. It resets the file to the state on development

Comment (aron4) [reply frontend/quickcheck/src/pages/ProjectOverview.jsx]: Ok, tried to do this. Please let me know if it works now, when you have time :)
